### ISSUE TITLE
Misleading default username when using docker template

### ISSUE THREAD

**Issue by abulte** [coder/coder#6457]
I tried to setup my first project using the example docker template, on a fresh install of Coder (v0.18.1+e3a4861).

I used a throwaway VPS to make some tests, so I did not bother to create a dedicated user and just used `root` (default Debian install) for everything.

While configuring coder, I used `coder login https://my.coder.com` from the VPS shell. Thus, the default username prompted was `root` and I just accepted it.

This led to the following error when creating my first project from the docker template:

```
Error: process "/bin/sh -c useradd --groups sudo --no-create-home --shell /bin/bash ${USER} \t&& echo \"${USER} ALL=(ALL) NOPASSWD:ALL\" >/etc/sudoers.d/${USER} \t&& chmod 0440 /etc/sudoers.d/${USER}" did not complete successfully: exit code: 9
```

After a while, I realised the `$USER` variable must be `root`, and adding `root` to sudoers does not really make sense... I fixed it by changing my username to something other than `root` and rebuilding the project.

I'm not sure what the ideal solution is:
- add somewhere in the QuickStart docs that using root for user creation is a bad idea?
- add a warning in the docker template doc?
- add a conditional to the example Dockerfile, if user is root do not do the sudo dance?

Anyway, putting it here since I've scratched my head a while and found no reference to this error. Hope this can help some people.

TLDR; don't use `root` as your coder username.
